src/Datadog.Trace.ClrProfiler.Managed/Emit/MethodBuilder.cs

```diff
@@ -421,6 +422,12 @@ private MethodInfo VerifyMethodFromToken(MethodInfo methodInfo)
                 return null;
             }
 
+            if (!methodInfo.IsStatic && !methodInfo.ReflectedType.IsAssignableFrom(_concreteType))
+            {
+                Log.Warning($"_concreteType cannot be assigned to the type containing the MethodInfo representing the instance method: {detailMessage}");
```

Comment:
nit: we should probably use structure logging here instead of interpolated strings. It can reduce GC pressure generally and Serilog is optimised for it. There's other log messages in this file that could do with the same treatment 🙂

Reply:
Do you mind if I make the logging changes in a separate PR? 😄

Reply:
Haha, of course!